MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/MainActivity.java

```diff
@@ -136,9 +151,46 @@ private void showAddBookDialogFragment() {
         addBookDialogFragment.show(fragmentManager, "New Entry");
     }
 
+    private void showSortItemsDialogFragment() {
+        Bundle args = new Bundle();
+        args.putString("sort", currentSortCriteria);
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        SortItemsDialogFragment sortItemsDialogFragment = new SortItemsDialogFragment();
+        sortItemsDialogFragment.setArguments(args);
```

Comment:
I would suggest to encapsulate the arguments passing directly into the fragment. What I mean by this is, in the `SortItemsDialogFragment.java`:

```java
private static final String EXTRA_SORT_CRITERIA = "com.izor066.android.mediatracker.EXTRA_SORT_CRITERIA";

public static SortItemsDialogFragment createWith(String currentSortCriteria) { // consider enum instead of string
    Bundle args = new Bundle();
    args.putString(EXTRA_SORT_CRITERIA, currentSortCriteria); // consider putSerializable() for enum
    SortItemsDialogFragment fragment = new SortItemsDialogFragment();
    fragment.setArguments(args);
    return fragment;
}

public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    // rest of the code
    currentSortCriteria = getArguments().getString(EXTRA_SORT_CRITERIA); //consider getSerializable() for enum
    // rest of the code
}
```

And now in `MainActivity.java`, in `showSortItemsDialogFragment()` you can call:

```java
SortItemsDialogFragment sortItemsDialogFragment = SortItemsDialogFragment.createWith(currentSortCriteria);
sortItemsDialogFragment.show(getSupportFragmentManager(), "sort");
```

This way only the fragment knows about its extra arguments and you keep this unknown to the external classes (`MainActivity`).